monkestation/code/modules/cassettes/cassette_approval.dm

```diff
@@ -131,8 +133,78 @@ GLOBAL_LIST_INIT(cassette_reviews, list())
 	message_admins("[submitter]'s tape has been approved by [user]")
 	logger.Log(LOG_CATEGORY_MUSIC, "[submitter]'s tape has been approved by [user]", list("approver" = user.name, "submitter" = submitter.name))
 	action_taken = TRUE
+	verdict = "APPROVED"
 
 /proc/fetch_review(id)
 	return GLOB.cassette_reviews[id]
 
 #undef ADMIN_OPEN_REVIEW
+
+// Handles UI to manage cassettes.
+/client/proc/review_cassettes() //Creates a verb for admins to open up the ui
+	set name = "Review Cassettes"
+	set desc = "Review this rounds cassettes."
+	set category = "Admin.Game"
+	var/datum/review_cassettes/tgui = new(usr)//create the datum
+	tgui.ui_interact(usr)//datum has a tgui component, here we open the window
+
+/datum/review_cassettes
+	var/client/holder //client of whoever is using this datum
+	var/is_funmin = FALSE
+
+/datum/review_cassettes/New(user)//user can either be a client or a mob due to byondcode(tm)
+	if (istype(user, /client))
+		var/client/user_client = user
+		holder = user_client //if its a client, assign it to holder
+	else
+		var/mob/user_mob = user
+		holder = user_mob.client //if its a mob, assign the mob's client to holder
+
+	is_funmin = check_rights(R_FUN)
+/datum/review_cassettes/ui_state(mob/user)
+	return GLOB.admin_state
+
+/datum/review_cassettes/ui_close()// Don't leave orphaned datums laying around. Hopefully this handles timeouts?
+	qdel(src)
+
+/datum/review_cassettes/ui_interact(mob/user, datum/tgui/ui) // Open UI and update as it remains open.
+	. = ..()
+	if(is_funmin)
+		ui = SStgui.try_update_ui(user, src, ui)
+		if(!ui)
+			ui = new(user, src, "CassetteManager")
+			ui.open()
+
+/datum/review_cassettes/ui_data(mob/user)
+	. = ..()
+	var/list/data = list("cassettes" = list()) // Initialize main data structure with an empty "cassettes" list
+
+	if(isnull(GLOB.cassette_reviews) || !GLOB.cassette_reviews.len)
+		return data
```

Comment:
```suggestion
```
unneeded

Reply:
As mentioned in another suggestion was getting runtimes without it.